mods/classic-taskbar-context-menu.wh.cpp

```diff
@@ -0,0 +1,591 @@
+	DestroyIcon(hIcon);
+
+	ExtractIconEx(L"shell32.dll", ICON_SHOWDESKTOP_INDEX, NULL, &hIcon, 1);
+	MyIcons[MYICON_SHOWDESKTOP] = IconToBitmap(hIcon, DPI_SCALE(16));
+	DestroyIcon(hIcon);
+}
+
+void MyIcons_Free() {
+	for (int i = 0; i < MyIcons_Count; i++) {
+		if (MyIcons[i] != NULL) DeleteObject(MyIcons[i]);
+    }
+}
+
+
+
+typedef struct MENU98_INIT_T {
+	HWND hWnd_TaskBar;
+	void* TrackPopupMenuEx;
+	LPSTR cmdLine;
+} MENU98_INIT;
+
+
+typedef DWORD(__cdecl* FPT___Menu98Init)(MENU98_INIT*);
+typedef DWORD(__cdecl* FPT___Menu98Unload)(LPVOID);
+
+
+
+static HMODULE gLibModule = 0;
+
+static LONG_PTR OldWndProc_TaskBar = NULL;
+static LONG_PTR OldWndProc_TaskBar_SecondScreen = NULL;
+
+static HWND hWnd_TaskBar = 0;
+static HWND hWnd_TaskBar_SecondScreen = NULL;
+static HWND hWnd_ATL1 = 0;
+static HWND hWnd_ATL2 = 0;
+
+static HMODULE menu98Module = 0;
+
+void ClassicMenu(HMENU hMenu) {
+	MENUINFO info;
+
+	info.cbSize = sizeof(MENUINFO);
+	info.fMask = MIM_BACKGROUND;
+
+	GetMenuInfo(hMenu, &info);
+    
+	info.hbrBack = NULL;
+
+	SetMenuInfo(hMenu, &info);
+
+	for (int i = 0; i < GetMenuItemCount(hMenu); i++) {
+		MENUITEMINFO menuInfo;
+
+		menuInfo.cbSize = sizeof(MENUITEMINFO);
+		menuInfo.fMask = MIIM_FTYPE | MIIM_SUBMENU;
+
+		GetMenuItemInfo(hMenu, i, true, &menuInfo);
+
+		menuInfo.fType &= ~MFT_OWNERDRAW;
+		menuInfo.fMask = MIIM_FTYPE;
+
+		SetMenuItemInfo(hMenu, i, true, &menuInfo);
+	}
+}
+
+void ClassicMenuIfPossible(HWND hWnd, HMENU hMenu) {
+	char clsName[256];
+	GetClassNameA(hWnd, clsName, 256);
+
+	if (strcmp(clsName, "TrayShowDesktopButtonWClass") == 0) {
+		if (settings.showIcons) SetMenuItemBitmaps(hMenu, 0x1A2D, MF_BYCOMMAND, MyIcons_Get(MYICON_SHOWDESKTOP), MyIcons_Get(MYICON_SHOWDESKTOP));
+		ClassicMenu(hMenu);
+	} else if (
+		strcmp(clsName, "NotificationsMenuOwner") == 0 ||		// Notification Button
+		strcmp(clsName, "LauncherTipWnd") == 0 ||				// Win+X menu
+		strcmp(clsName, "MultitaskingViewFrame") == 0 ||		// Multitask Button
+		
+        hWnd == hWnd_ATL1 || hWnd == hWnd_ATL2) {				// Network Icon and Volumn Icon
+		ClassicMenu(hMenu);
+
+		for (int i = 0; i < GetMenuItemCount(hMenu); i++)
+			ClassicMenu(GetSubMenu(hMenu, i));
+	}
+}
+
+void RestoreWndProc() {
+	if (OldWndProc_TaskBar != NULL) {
+		SetWindowLongPtr(hWnd_TaskBar, GWLP_WNDPROC, OldWndProc_TaskBar);
+    }
+
+	if (OldWndProc_TaskBar_SecondScreen != NULL) {
+		SetWindowLongPtr(hWnd_TaskBar_SecondScreen, GWLP_WNDPROC, OldWndProc_TaskBar_SecondScreen);
+    }
+}
+
+void CloseBackground() {
+	MyIcons_Free();
+	RestoreWndProc();
+
+	if (menu98Module != NULL) {
+		FPT___Menu98Unload fpMenu98Unload = (FPT___Menu98Unload)GetProcAddress(menu98Module, "__Menu98Unload");
+
+		if (fpMenu98Unload != NULL) {
+			fpMenu98Unload(NULL);
+		} else {
+			MessageBoxW(0, L"Failed to unload Menu98!", L"Fatal Error", MB_ICONERROR);
+		}
+	}
+
+
+	CloseHandle(CreateThread(NULL, 0, (LPTHREAD_START_ROUTINE)FreeLibraryAndExitThread, gLibModule, 0, NULL));
```

Comment:
Looks good overall. Remove this, Windhawk will unload this mod, no need to do that manually and it can lead to problems.